### PR TITLE
test(web): improve test coverage and edge cases

### PR DESCRIPTION
**Pull request by ShahdAshraff**

### CODE REVIEW

**Comment by ashmod** [comment]
@ShahdAshraff Thanks for working on this. I think this is moving in a better direction in a few places, especially the added malformed graph payload cases in `graphIO.test.ts` and the “already retried” auth case in `setupClientInterceptors.test.ts`.

That said, I don’t think this PR is quite the web test refactor we were aiming for yet. Most of the changes are still small updates to existing utility-level tests, like `initials.test.ts`, `permissions.test.ts`, `executionConversion.test.ts`, and `graphIO.test.ts`. 

Renaming tests and adding a few edge cases helps, but the sprint goal was more about **shifting coverage toward behavior a *real* user could hit**.

Could you take another pass and add/refactor tests around actual web behavior? For example:
- in `apps/web/src/lib/permissions.test.ts`, instead of only asserting helper booleans, cover a realistic permission scenario such as viewer/editor/owner access to workflow actions
- in `apps/web/src/features/canvas/lib/graphIO.test.ts`, use a realistic multi-node workflow import/export case and assert that invalid/unsafe graph state is dropped without losing valid branches
- in `apps/web/src/features/canvas/lib/executionConversion.test.ts`, reduce the mocked-helper style and cover how a realistic execution graph with success/error/partial node states becomes frontend execution state
- in `apps/web/src/lib/api/setupClientInterceptors.test.ts`, keep the useful retry case, but frame/add cases around the user-visible auth behavior: refresh succeeds, refresh fails, redirect path is preserved, already-retried requests do not loop

Also, please look beyond the files already touched here and identify web areas that are not covered yet, or that clearly need deeper tests. For example, workflow list/actions, canvas editing flows, sharing/permission UI, execution status rendering, auth/session expiry behavior, and form error states.

The bar I’m looking for is: “could this test catch a regression that would become a real user bug report?” A few of the additions do that, but overall this PR still feels more like cleanup of existing tests than the requested behavior-focused web refactor.

Please let me know if you need further clarification on this!

File: apps/web/src/features/canvas/lib/executionConversion.test.ts [hunk 1]
```diff
@@ -1,31 +1,9 @@
...
-vi.mock("./graphIO", () => ({
-  sanitizeGraph: sanitizeGraphMock,
-}));
-
-vi.mock("./autoLayout", () => ({
-  applyAutoLayout: applyAutoLayoutMock,
-}));
+import { describe, expect, it } from "vitest";
 
 import { extractGraphSnapshot, rtesDocToExecutionState } from "./executionConversion";
```
[the remark this answers is not in the document]
Comment (ashmod) [reply apps/web/src/features/canvas/lib/executionConversion.test.ts:3]: This is because of the current merge conflict, please resolve by keeping main's autolayout tests @ShahdAshraff

File: apps/web/src/lib/initials.test.ts [hunk 2]
```diff
@@ -2,68 +2,34 @@ import { describe, it, expect } from "vitest";
...
-  it("handles arabic characters", () => {
+  it("test_non_latin_names_use_first_and_last_characters", () => {
     expect(getInitials("أحمد محمد")).toBe("أم");
   });
 
-  it("handles names with non-alphabetic characters", () => {
+  it("test_punctuation_inside_tokens_does_not_split_name_parts", () => {
     expect(getInitials("John-Doe")).toBe("J");
     expect(getInitials("John O'Connor")).toBe("JO");
   });
```
[the remark this answers is not in the document]
Comment (ashmod) [reply apps/web/src/lib/initials.test.ts]: Good catch. Please ensure naming consistency across the frontend test suite @ShahdAshraff

**Review by ashmod — APPROVED** [review]
